### PR TITLE
Update the PR template

### PR DESCRIPTION
**Pull request by georgepisaltu**
# Reason for This PR

We want our discussions to happen on issues and then have contributions result from a clear conclusion on said issues. PRs opened without backing issues are more likely to be closed or completely refactored, leading to wasted work from the contributor. 

## Description of Changes

Updated the PR template to ensure PRs have a backing issue where the proposed solution has already been established and all relevant discussion took place.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Reason for This PR
 
-`[Author TODO: add issue # or explain reasoning.]`
+`[Author TODO: add issue #.]`
+`[If there is no issue which states the need for this PR, create one first.]`
```
Comment (gc-plp) [inline .github/pull_request_template.md:4]: I find that this statement is a bit disconnected with what's required in the checkbox below: `The issue which led to this PR has a clear conclusion`.

I'd rephrase this as something along the lines of `If there is no issue which states the need for this PR, create one and continue the PR after the issue has a clear conclusion.`.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -19,7 +21,9 @@ the terms of the Apache 2.0 license.
 `[Reviewer TODO: Verify that these criteria are met. Request changes if not]`
 
 - [ ] All commits in this PR are signed (`git commit -s`).
-- [ ] The reason for this PR is clearly provided (issue no. or explanation).
+- [ ] The reason for this PR is clearly provided (issue no.).
```
Comment (gc-plp) [inline .github/pull_request_template.md:24]: I think that we're adding too many checkboxes here. The last two checkboxes can be merged into one.

File: .github/pull_request_template.md (hunk 1 above)
Comment (georgepisaltu) [reply .github/pull_request_template.md:4]: I think a big benefit would be to have contributors NOT open the PR until they both create an issue and reach a conclusion there. Creating a PR and an issue at the same time could likely lead to wasted work from the contributor, as there may be an entirely different approach agreed on the issue, and the initial PR variant has to be scrapped.

That being said, we could rephrase to `Open the PR after the related issue has a clear conclusion. If there is no issue which states the need for this PR, create one first.`

This phrasing also covers the case where an issue was opened, but it has no conclusion yet.

File: .github/pull_request_template.md (hunk 2 above)
Comment (georgepisaltu) [reply .github/pull_request_template.md:24]: I will remove the `The reason for this PR is clearly provided (issue no.).` checkbox entirely as it is covered by the next one.

File: .github/pull_request_template.md (hunk 1 above)
Comment (gc-plp) [reply .github/pull_request_template.md:4]: > That being said, we could rephrase to Open the PR after the related issue has a clear conclusion. If there is no issue which states the need for this PR, create one first.

Sounds good :)

**Review by alsrdn — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]
